### DISCUSSION TITLE
Styling swiper buttons using Web-Components

### DISCUSSION THREAD

**Discussion by mohamed-cherifi** [nolimits4web/swiper#6547] [Q&A] [answered]
Hi, I've been trying to override the styling for the swiper buttons. But it seems I am not doing it right.
I tried updating the `styles.css` file with the following:
```
.swiper-button-next,
.swiper-button-prev {
  color: red !important;
}
```
it didn't work 😢 
I am using Swiper Element with React.

Any ideas?

**Comment by nolimits4web** [comment, marked as answer]
You can style web components only using css variables

**Reply by mohamed-cherifi** [reply]
so I can't override anything that's not one of [these properties](https://swiperjs.com/swiper-api#navigation-css-custom-properties)?

**Reply by nolimits4web** [reply]
You can't, but you can create your own buttons and style them as you wish

**Reply by mohamed-cherifi** [reply]
how would I go about doing that?

**Reply by mohamed-cherifi** [reply]
what if I use [injectStyles](https://swiperjs.com/element#injecting-styles)?

**Reply by Koreanderson** [reply]
@mohamed-cherifi 

You could create your own styled buttons in your component, then add attributes to your `swiper-container` web component for `navigation-next-el` and `navigation-prev-el`. The values will be selectors for your styled buttons.

```
<swiper-container
      navigation-next-el="#my-next-button"
      navigation-prev-el="#my-prev-button"
>
...
</swiper-container>
<button id="my-next-button">Next</button>
<button id="my-prev-button">Prev</button>
```

**Reply by mohamed-cherifi** [reply]
that worked
thank you!

**Reply by bielvelozo** [reply]
can you share your code? I have the same problem but it doesn't work for me

**Reply by SewaB** [reply]
Try using parameter:

injectStyles: [
    `
      .swiper-pagination {
        position: unset;
      }
    `
  ]

**Reply by jpierson-at-riis** [reply]
@SewaB, what was your suggestion meaning to fix?

**Reply by ZeroRWV** [reply]
I tried to use this approach, but it does not add a event listener on the buttons, so it doesn't work. I was assuming the swiper element js would handle it. Am I missing something?

**Reply by microrony** [reply]
Anyone trying this, try to set navigation to true as well. So it will be

```html
<swiper-container
      navigation-next-el="#my-next-button"
      navigation-prev-el="#my-prev-button"
      navigation="true"
>
...
</swiper-container>
<button id="my-next-button">Next</button>
<button id="my-prev-button">Prev</button>
```
